tests/plugins/keyword_test.py

```diff
@@ -12,24 +13,137 @@ class TestKeywordDetector(object):
+            ),
+            (
+                'theapikey:hopenobodyfinds>-_$#thisone'
+            ),
+            (
+                'apikey: "hopenobodyfinds>-_$#thisone"'
+            ),
+            (
+                "apikey:  'hopenobodyfinds>-_$#thisone'"
+            ),
+            # FOLLOWED_BY_EQUAL_SIGNS_RE
+            (
+                'my_password=hopenobodyfinds>-_$#thisone'
+            ),
+            (
+                'my_password= hopenobodyfinds>-_$#thisone'
+            ),
+            (
+                'my_password =hopenobodyfinds>-_$#thisone'
+            ),
+            (
+                'my_password_for_stuff = hopenobodyfinds>-_$#thisone'
+            ),
+            (
+                'my_password_for_stuff =hopenobodyfinds>-_$#thisone'
+            ),
+            (
+                'passwordone=hopenobodyfinds>-_$#thisone\n'
+            ),
+            (
+                'passwordone= "hopenobodyfinds>-_$#thisone"\n'
+            ),
+            (
+                'passwordone=\'hopenobodyfinds>-_$#thisone\'\n'
+            ),
+            # FOLLOWED_BY_QUOTES_AND_SEMICOLON_RE
+            (
+                'apikey "hopenobodyfinds>-_$#thisone";'  # Double-quotes
+            ),
+            (
+                'fooapikeyfoo "hopenobodyfinds>-_$#thisone";'  # Double-quotes
+            ),
+            (
+                'fooapikeyfoo"hopenobodyfinds>-_$#thisone";'  # Double-quotes
+            ),
+            (
+                'private_key \'hopenobodyfinds>-_$#thisone\';'  # Single-quotes
+            ),
+            (
+                'fooprivate_keyfoo\'hopenobodyfinds>-_$#thisone\';'  # Single-quotes
+            ),
+            (
+                'fooprivate_key\'hopenobodyfinds>-_$#thisone\';'  # Single-quotes
             ),
         ],
     )
-    def test_analyze(self, file_content):
+    def test_analyze_positives(self, file_content):
         logic = KeywordDetector()
 
         f = mock_file_object(file_content)
         output = logic.analyze(f, 'mock_filename')
         assert len(output) == 1
         for potential_secret in output:
             assert 'mock_filename' == potential_secret.filename
-        generated = list(logic.secret_generator(file_content))
-        assert len(generated) == len(output)
+            assert (
+                potential_secret.secret_hash
+                == PotentialSecret.hash_secret('hopenobodyfinds>-_$#thisone')
+            )
+
+    @pytest.mark.parametrize(
+        'file_content',
+        [
+            # FOLLOWED_BY_COLON_RE
+            (
+                'private_key "";'  # Nothing in the quotes
+            ),
+            (
+                'private_key \'"no spaces\';'  # Has whitespace in the secret
+            ),
+            (
+                'private_key "fake";'  # 'fake' in the secret
+            ),
+            (
+                'private_key "hopenobodyfindsthisone\';'  # Double-quote does not match single-quote
```

Comment:
Alright, so, I ought to fix this one so that it catches it.

It's a `Nit:` but 🤷‍♂️

Reply:
Decided not to, b/c it's not valid syntax in php anyhow, apparently